pythran/pythonic/random/shuffle.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef PYTHONIC_RANDOM_SHUFFLE_HPP
+#define PYTHONIC_RANDOM_SHUFFLE_HPP
+
+#include "pythonic/include/random/shuffle.hpp"
+
+#include "pythonic/utils/proxy.hpp"
+#include "pythonic/random/random.hpp"
+
+#include <limits>
+
+namespace pythonic {
+
+    namespace random {
+
+        template <class T>
+        void shuffle(T & seq) {
+            std::shuffle(seq.begin(), seq.end(), __random_generator);
+        }
+
+        template <class T, class function>
+        void shuffle(T & seq, function && randf) {
+            struct URG {
+                URG(function && f) : randf(randf) {}
+
+                typedef unsigned result_type;
+                result_type min() { return 0; }
+                /* -1 because of the floor() operation in the operator () */
```

Comment:
I meant the implicit truncation in the float-to-integer conversion. For example `5*0.99` will give `4`, so `max()` will be `5 - 1`.